util/fiber_socket_base.cc

```diff
@@ -116,12 +118,16 @@ error_code LinuxSocketBase::ListenUDS(const char* path, unsigned backlog) {
   addr.sun_family = AF_UNIX;
   memcpy(addr.sun_path, path, len);
   addr.sun_path[len] = 0;
-
-  return Listen((struct sockaddr*)&addr, sizeof(addr), backlog);
+  error_code ec = Bind((struct sockaddr*)&addr, sizeof(addr));
+  if (ec)
+    return ec;
+  ec = Chmod(path, permissions);
```

Comment:
calling `chmod` directly is better imho. sockets should not have Chmod operation.